docker/src/test/java/org/jclouds/docker/compute/SshToCustomPortLiveTest.java

```diff
@@ -0,0 +1,223 @@
+               runAsRoot(false).wrapInInitScript(false));
+         assertThat(response.getOutput().trim()).endsWith("hello");
+      } finally {
+         if (nodeId != null)
+            view.getComputeService().destroyNode(nodeId);
+      }
+   }
+
+   /**
+    * Build a new image with 2 tags on it in the test preparation phase.
+    * 
+    * @see org.jclouds.apis.BaseContextLiveTest#setupContext()
+    */
+   @Override
+   @BeforeClass(groups = { "integration", "live" })
+   public void setupContext() {
+      super.setupContext();
+
+      final String tag = toTag(IMAGE_REPOSITORY, IMAGE_TAG_1);
+      BuildOptions options = BuildOptions.Builder.tag(tag).verbose(false).nocache(false);
+      InputStream buildImageStream;
+      try {
+         buildImageStream = api().getMiscApi().build(BaseDockerApiLiveTest.tarredDockerfile(), options);
+         consumeStreamSilently(buildImageStream);
+      } catch (IOException e) {
+         e.printStackTrace();
+      }
+      image = api().getImageApi().inspectImage(tag);
+      api().getImageApi().tagImage(image.id(), IMAGE_REPOSITORY, IMAGE_TAG_2, true);
+   }
+
+   /**
+    * After the test remove created image (with all tags).
+    * 
+    * @see #setupContext()
+    */
+   @AfterClass
+   protected void tearDown() {
+      consumeStreamSilently(api().getImageApi().deleteImage(toTag(IMAGE_REPOSITORY, IMAGE_TAG_1)));
+      consumeStreamSilently(api().getImageApi().deleteImage(toTag(IMAGE_REPOSITORY, IMAGE_TAG_2)));
+   }
+
+   /**
+    * Configure used modules. A custom {@link LoginPortForContainer} binding is
+    * added among logging and SSH module.
+    * 
+    * @see org.jclouds.compute.internal.BaseGenericComputeServiceContextLiveTest#setupModules()
+    */
+   @Override
+   protected Iterable<Module> setupModules() {
+      return ImmutableSet.<Module> of(getLoggingModule(), new SshjSshClientModule(), new AbstractModule() {
+         @Override
+         protected void configure() {
+            bind(LoginPortForContainer.class).toInstance(new LoginPortForContainer() {
+               @Override
+               public Optional<Integer> apply(Container input) {
+                  return Optional.of(SSH_PORT);
+               }
+            });
+         }
+      });
+   }
+
+   /**
+    * Return DockerApi for current Context.
+    * 
+    * @return
+    */
+   private DockerApi api() {
+      return view.unwrapApi(DockerApi.class);
+   }
+
+   /**
+    * Read all data from given InputStream and throw away all the bits.
+    * 
+    * @param is
+    */
+   private static void consumeStreamSilently(InputStream is) {
+      char[] tmpBuff = new char[8 * 1024];
+      // throw everything away
+      InputStreamReader isr = new InputStreamReader(is);
+
+      try {
+         try {
+            while (isr.read(tmpBuff) > -1) {
+               // empty
+            }
+         } finally {
+            isr.close();
+         }
+      } catch (IOException e) {
+         e.printStackTrace();
```

Comment:
Propagate the exception or use the `java.util.Logger.getAnonymousLogger()` to log test errors.